Add code for parsing journal superblocks

nicholasbishop opened the pull request:
This is `expect(unused)` until more of the journal code is added.

tedbrandston commented on src/journal/superblock.rs:
```diff
@@ -0,0 +1,260 @@
+            JournalSuperblock::read_bytes(&block).unwrap_err(),
+            Incompatible::JournalIncompatibleFeatures(0x10),
+        );
+    }
```
nit: could optionally add a test for unsupported incompat features?

nicholasbishop replied:
done

nicholasbishop replied:
or, not done. good thing you asked for this, as I'm realizing I updated the test incompletely and it still passes, revealing that `bitflags` doesn't behave quite like I thought it did.

nicholasbishop replied:
OK now it is fixed properly.

tedbrandston commented on src/journal/superblock.rs:
```diff
@@ -0,0 +1,260 @@
+        write_u32be(&mut block, SUPERBLOCK_FEATURE_INCOMPAT_OFFSET, 0x10);
+        assert_eq!(
+            JournalSuperblock::read_bytes(&block).unwrap_err(),
+            Incompatible::JournalIncompatibleFeatures(0x10),
```
nit: This error name with this value feels misleading to me. I guess I'm just reading it wrong: "here's the features that are wrong", rather than "there's an error related to this set of 'incompatible features'". I might just be complaining about the existing terminology.

I imagine in the future this will evolve towards the missing/unknown/incompatible split used elsewhere?

nicholasbishop commented:
I've reworked this slightly in response to your comments. `Incompatible::JournalIncompatibleFeatures` has been replaced by two variants: `Incompatible::MissingRequiredJournalFeatures` and `Incompatible::UnsupportedJournalFeatures`. I think this is a little clearer than the unknown/missing/incompatible variants used for the features in the main superblock, and might make the same change there.

Added four commits on top of the original change, so requesting a re-review.
